web/features/search/SearchContext.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  useCallback,
+} from "react";
+import { getClient } from "../../utils/sanity/sanity.server";
+
+export type SearchResultType = {
+  slug: {
+    current: string;
+  };
+  title: string;
+  availableFrom: string;
+};
+
+const SearchContext = createContext(null);
+
+type SearchContextType = {
+  searchStr: string;
+  setSearchStr: (searchStr: string) => void;
+  loading: boolean;
+  searchResults: SearchResultType[];
+  searchIsActive: boolean;
+  search: () => void;
+  closeSearch: () => void;
+};
+
+export const useSearch = (): SearchContextType => {
+  const context: SearchContextType | null = useContext(SearchContext);
+  if (context === null) {
+    throw new Error("useSearch must be used within a SearchProvider");
+  }
+  return {
+    searchStr: context.searchStr,
+    setSearchStr: context.setSearchStr,
+    loading: context.loading,
+    searchResults: context.searchResults,
+    searchIsActive: context.searchIsActive,
+    search: context.search,
+    closeSearch: context.closeSearch,
+  };
```

Comment:
Her kan du spare deg noen linjer: 

```suggestion
  return context;
```